tracer/src/Datadog.Trace/Iast/Propagation/PropagationModuleImpl.cs

```diff
@@ -105,10 +105,10 @@ public static void AddTainted(string? input, Source source)
 
             var taintedObjects = iastContext.GetTaintedObjects();
 
-            if (PropagateResultWhenInputTainted(result, firstInput, taintedObjects) ||
-                PropagateResultWhenInputTainted(result, secondInput, taintedObjects) ||
-                PropagateResultWhenInputTainted(result, thirdInput, taintedObjects) ||
-                PropagateResultWhenInputTainted(result, fourthInput, taintedObjects))
+            if (PropagateResultWhenInputTainted(result!, firstInput, taintedObjects) ||
+                PropagateResultWhenInputTainted(result!, secondInput, taintedObjects) ||
+                PropagateResultWhenInputTainted(result!, thirdInput, taintedObjects) ||
+                PropagateResultWhenInputTainted(result!, fourthInput, taintedObjects))
```

Comment:
it's a minor point, but it looks a bit strange to change `result` to a nullable string then assert that it isn't null at each usage. It seems a better solution would be to change the nullability of result in the overload of `PropagateResultWhenInputArrayTainted` that's called. It seems we expect that `result` could be null as one of the first things we do is call `string.IsNullOrEmpty`

Reply:
I was thinking that, in terms of performance, checking before could be better. Also, PropagateResultWhenInputTainted does not check for null values in result. PropagateResultWhenInputArrayTainted does, but it is not called.